github/Issue.py

```diff
@@ -347,119 +316,100 @@ def create_comment(self, body):
+        state: Opt[str] = NotSet,
+        milestone: Opt[Milestone] = NotSet,
+        labels: Opt[list[str]] = NotSet,
+        assignees: Opt[list[str]] = NotSet,
+        state_reason: Opt[str] = NotSet,
     ):
         """
         :calls: `PATCH /repos/{owner}/{repo}/issues/{number} <https://docs.github.com/en/rest/reference/issues>`_
-        :param title: string
-        :param body: string
-        :param assignee: string or :class:`github.NamedUser.NamedUser` or None
-        :param state: string
-        :param milestone: :class:`github.Milestone.Milestone` or None
-        :param labels: list of string
-        :param assignees: list of string or :class:`github.NamedUser.NamedUser`
-        :param state_reason: string
-        :rtype: None
-        """
-        assert title is github.GithubObject.NotSet or isinstance(title, str), title
-        assert body is github.GithubObject.NotSet or isinstance(body, str), body
-        assert (
-            assignee is github.GithubObject.NotSet
-            or assignee is None
-            or isinstance(assignee, github.NamedUser.NamedUser)
-            or isinstance(assignee, str)
+        """
+        assert isinstance(title, (str, _NotSetType)), title
+        assert isinstance(body, (str, _NotSetType)), body
+        assert assignee is None or isinstance(
+            assignee, (github.NamedUser.NamedUser, str, _NotSetType)
         ), assignee
-        assert assignees is github.GithubObject.NotSet or all(
-            isinstance(element, github.NamedUser.NamedUser) or isinstance(element, str)
+        assert isinstance(assignees, _NotSetType) or all(
+            isinstance(element, (github.NamedUser.NamedUser, str))
             for element in assignees
         ), assignees
-        assert state is github.GithubObject.NotSet or isinstance(state, str), state
-        assert (
-            milestone is github.GithubObject.NotSet
-            or milestone is None
-            or isinstance(milestone, github.Milestone.Milestone)
+        assert isinstance(state, (_NotSetType, str)), state
+        assert milestone is None or isinstance(
+            milestone, (_NotSetType, github.Milestone.Milestone)
         ), milestone
-        assert labels is github.GithubObject.NotSet or all(
+        assert isinstance(labels, _NotSetType) or all(
             isinstance(element, str) for element in labels
         ), labels
-        post_parameters = dict()
-        if title is not github.GithubObject.NotSet:
-            post_parameters["title"] = title
-        if body is not github.GithubObject.NotSet:
-            post_parameters["body"] = body
-        if assignee is not github.GithubObject.NotSet:
+
+        post_parameters = NotSet.remove_unset_items(
+            {
+                "title": title,
+                "body": body,
+                "state": state,
+                "state_reason": state_reason,
+                "labels": labels,
+            }
```

Comment:
`remove_unset_items` will remove `assigne` if it is `NotSet`:
```suggestion
                "assignee": assignee._identity if isinstance(assignee, NamedUser) else assignee
            }
```